library/src/main/java/com/getstream/sdk/chat/view/MessageListView.java

```diff
@@ -269,6 +365,36 @@ private void initAdapter() {
         this.setMessageListItemAdapter(adapter);
     }
 
+    public void setScrollButtonBehaviour(ScrollButtonBehaviour scrollButtonBehaviour) {
+        this.scrollButtonBehaviour = scrollButtonBehaviour;
+    }
+
+    public void setNewMessagesBehaviour(int newMessagesBehaviour) {
```

Comment:
Could we use an enum for this instead when we're dealing with it in code? I know we also need the int handling for the XML attributes, but it would be nicer to deal with ints only in that part.

Reply:
I thought about `enum`, but I'm not sure if we should use it or not in our libraries as it increases the dex much more than I simple constant: https://medium.com/better-programming/android-how-enums-can-impact-the-performance-f787ef5903dd.

Since this is a SDK that others will use, I think we should avoid enums to create the smallest library as possible and go easier on memory as possible. What do you think?

Reply:
For a while now, the official recommendation on this was that enums are fine to use (e.g. [this bit from I/O 2018](https://youtu.be/IrMw7MEgADk?t=849)), especially in application level code. They're using it more and more in the framework as well, and we're much closer to writing application code here than a tight inner loop of the framework anyway.

Overall, the safety gained here from having a well-typed API seems to vastly outweigh the possible performance gains.